WooCommerce/Classes/ViewRelated/Orders/Refund/RefundSubmissionUseCase.swift

```diff
@@ -0,0 +1,365 @@
+        cardPresentRefundOrchestrator.cancelRefund { _ in
+            // TODO: 5984 - tracks in-person refund cancellation
+        }
+    }
+
+    /// Submits the refund to the site.
+    /// - Parameters:
+    ///   - refund: the refund to submit.
+    ///   - onCompletion: called when the submission completes.
+    func submitRefundToSite(refund: Refund, onCompletion: @escaping (Result<Void, Error>) -> Void) {
+        let action = RefundAction.createRefund(siteID: details.order.siteID, orderID: details.order.orderID, refund: refund) { [weak self] _, error  in
+            guard let self = self else { return }
+            if let error = error {
+                DDLogError("Error creating refund: \(refund)\nWith Error: \(error)")
+                self.trackCreateRefundRequestFailed(error: error)
+                return onCompletion(.failure(error))
+            }
+            onCompletion(.success(()))
+            self.trackCreateRefundRequestSuccess()
+        }
+        stores.dispatch(action)
+        trackCreateRefundRequest()
+    }
+}
+
+// MARK: - Analytics
+private extension RefundSubmissionUseCase {
+    /// Tracks when the create refund request is made.
+    func trackCreateRefundRequest() {
+        analytics.track(event: WooAnalyticsEvent.IssueRefund.createRefund(orderID: details.order.orderID,
+                                                                          fullyRefunded: details.amount == details.order.total,
+                                                                          method: .items,
+                                                                          gateway: details.order.paymentMethodID,
+                                                                          amount: details.amount))
+    }
+
+    /// Tracks when the create refund request succeeds.
+    func trackCreateRefundRequestSuccess() {
+        analytics.track(event: WooAnalyticsEvent.IssueRefund.createRefundSuccess(orderID: details.order.orderID))
+    }
+
+    /// Tracks when the create refund request fails.
+    func trackCreateRefundRequestFailed(error: Error) {
+        analytics.track(event: WooAnalyticsEvent.IssueRefund.createRefundFailed(orderID: details.order.orderID, error: error))
+    }
+}
+
+// MARK: Connected Card Readers
+private extension RefundSubmissionUseCase {
+    func observeConnectedReadersForAnalytics() {
+        let action = CardPresentPaymentAction.observeConnectedReaders() { [weak self] readers in
+            self?.connectedReader = readers.first
+        }
+        stores.dispatch(action)
+    }
+}
+
+// MARK: Definitions
+private extension RefundSubmissionUseCase {
+    /// Mailing a receipt failed but the SDK didn't return a more specific error
+    ///
+    enum RefundSubmissionError: Error {
+        case invalidRefundAmount
+        case unknownPaymentGatewayAccount
+    }
+
+    enum Localization {
+        private static let collectPaymentWithoutName = NSLocalizedString("Refund payment",
```

Comment:
static names here should be for refund not payment

Reply:
oops my bad! updated in https://github.com/woocommerce/woocommerce-ios/pull/6602/commits/b2446b397f9e816c1cb122a52db54fc7ca9f9904